### ISSUE TITLE
Applet broken with LWJGL 2.9.2, at least on some OSes

### ISSUE THREAD

**Issue by Cervator** [MovingBlocks/FacadeApplet#18]
After the 2.9.2 LWJGL upgrade I get the following error:

```
Exception in thread "Thread-63" java.lang.NoClassDefFoundError: org/lwjgl/LWJGLUtil
	at org.lwjgl.util.applet.AppletLoader$4.findLibrary(AppletLoader.java:1224)
	at java.lang.ClassLoader.loadLibrary(Unknown Source)
	at java.lang.Runtime.loadLibrary0(Unknown Source)
	at java.lang.System.loadLibrary(Unknown Source)
	at org.terasology.rendering.oculusVr.OculusVrHelper.loadNatives(OculusVrHelper.java:67)
	at org.terasology.utilities.LWJGLHelper.initOculus(LWJGLHelper.java:68)
	at org.terasology.utilities.LWJGLHelper.initNativeLibs(LWJGLHelper.java:44)
	at org.terasology.engine.subsystem.lwjgl.BaseLwjglSubsystem.preInitialise(BaseLwjglSubsystem.java:40)
	at org.terasology.engine.subsystem.lwjgl.LwjglGraphics.preInitialise(LwjglGraphics.java:94)
	at org.terasology.engine.TerasologyEngine.preInitSubsystems(TerasologyEngine.java:257)
	at org.terasology.engine.TerasologyEngine.<init>(TerasologyEngine.java:157)
	at org.terasology.engine.TerasologyApplet$2.run(TerasologyApplet.java:144)
```

I don't know why Oculus is in there, I suspect that's a red herring

You can run the applet fine with the engine having updated to 2.9.2 - it works if you take the 2.9.2 applet dir and overwrite just the lwjgl_util_applet-2.9.2.jar with the 2.9.1 version, but then that version has the hack we were trying to remove.

I wonder if 2.9.2 will fail even with a hacked util_applet jar, I suspect so, but haven't tested. For the fun of it I did try a different hack in *just* downgrading the util_applet jar to 2.9.1, leaving everything else 2.9.2, then signing it all, but as expected then we get the old error.

Despite having labelled 2.9.2 the final ever LWJGL 2 it looks like they've updated to 2.9.3 on GitHub and a few more commits have made it in. I wonder if some bug killed applets in general (or if anybody would notice)

@msteiger would you mind taking a look? 

@emanuele3d I added a quick tweak so the applet can be built locally now without the signer stuff - of course it won't be signed but you can see the end product :-)

**Comment by msteiger** [comment]
Yes, the Oculus code is just the first thing to call LWJGL code. It seems like the main `lwjgl.jar` is not on the classpath.

Looking at the [index.html file](http://jenkins.terasology.org/job/TerasologyApplet/lastSuccessfulBuild/artifact/build/distributions/applet/index.html), everything looks ok, though. 

Also, the LWJGLUtil class seems to be almost identical to the 2.9.1 version. Long story short, I have no clue what could be behind this :cry: 

So, if we revert cc681ae62f98053a5, it will work again?

Is there some kind of access tracker that could tell us how often the applet is actually used?

**Comment by emanuele3d** [comment]
Thanks Cervator for removing the signer stuff! I'll check it out at some
point!!

>

**Comment by Cervator** [comment]
@msteiger - I suspect we could use your old hack + my new hack, so we're running mostly 2.9.2, excepting the applet util file, which probably isn't used for anything other than actually launching the applet. So at least the game will run at 2.9.2 (which it probably will anyway since the engine is on 2.9.2 for sure)

That's my theory, anyway, I could be off - just haven't tested that yet. Want to do so if you get a chance?

I've got Google Analytics hooked into the applet index.html among other things. It averages about 100-150 hits daily, a small spike about a week ago got us past 400 for a day then it remains a little elevated since, although it sadly doesn't work now :D

I was more wondering if LWJGL themselves do any thorough applet testing for patch releases.

@emanuele3d - np, you reminded me it has to be done. It is just conditionally enabled anyway, not removed :-)

**Comment by Cervator** [comment]
Fixed with new hack + partial old hack. I rolled back *just* the applet util jar, which solely contains the loader, and it works. All other jars are still 2.9.1 and only the applet util jar gets the hack so it'll work at that version.

So everything effectively should be at 2.9.2 and the hack is minimal. Would be nice to be fully rid of it but it isn't worth the effort right now unless somebody gets bored. Maybe if a 2.9.3 comes out or whenever we go to the next big thing. There is still a TODO in place in code which is good enough, so closing this.
